### PR TITLE
HSEARCH-3136  Throw an exception when an element of @IndexedEmbedded.includePaths did not match anything

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3136

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/ConfiguredIndexSchemaNestingContext.java [hunk 1]
```diff
@@ -91,6 +93,11 @@ public String toString() {
 			while ( nextDotIndex >= 0 ) {
 				String objectName = prefixToParse.substring( afterPreviousDotIndex, nextDotIndex );
 				contextBuilder.appendObject( objectName );
+
+				// Make sure to mark the paths as encountered in the filter
+				String objectNameRelativeToFilter = prefixToParse.substring( 0, nextDotIndex );
+				filter.isPathIncluded( objectNameRelativeToFilter );
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/ConfiguredIndexSchemaNestingContext.java]: I would use the name `markPathIncluded` or `markPathAsIncluded`.

**Comment by gsmet** [comment]
Oh and it needs a rebase.

File: engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/ConfiguredIndexSchemaNestingContext.java (hunk 1 above)
Comment (yrodiere) [reply engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/ConfiguredIndexSchemaNestingContext.java]: The thing is, this method returns a boolean telling you whether the path is included or not. Here's I'm just twisting it: I know it has a side effect, which is marking the path as encountered, so I use it just for the side effect.

Every where else we really mainly use the method to know whether the path is included or not.

I'll add a comment to clarify.

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**